src/nvim/file.c

```diff
@@ -0,0 +1,293 @@
+/// @file file.c
+///
+/// Buffered reading/writing to a file. Unlike fileio.c this is not dealing with
+/// Neovim stuctures for buffer, with autocommands, etc: just fopen/fread/fwrite
+/// replacement.
```

Comment:
This should go in `os/`, no? That is where "[relatively] application-agnostic" operations live.